lib/models/room-state.js

```diff
@@ -181,10 +188,41 @@ RoomState.prototype.getLastModifiedTime = function() {
 };
 
 /**
+ * Get user IDs with the specified display name.
+ * @param {string} displayName The display name to get user IDs from.
+ * @return {string[]} An array of user IDs or an empty array.
+ */
+RoomState.prototype.getUserIdsWithDisplayName = function(displayName) {
+    return this._displayNameToUserIds[displayName] || [];
+};
+
+/**
  * The RoomState class.
  */
 module.exports = RoomState;
 
+
+function _updateDisplayNameCache(roomState, userId, displayName) {
+    var oldName = roomState._userIdsToDisplayNames[userId];
+    delete roomState._userIdsToDisplayNames[userId];
+    if (oldName) {
+        var existing = roomState._displayNameToUserIds[oldName] || [];
+        for (var i = 0; i < existing.length; i++) {
+            if (existing[i] === userId) {
+                existing.splice(i, 1);
```

Comment:
might be worth a comment spelling out the .splice() idiom for deleting the matching old entry from the array - i had to think about this a bit to see what was going on.